hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/StreamEventJournalP.java

```diff
@@ -0,0 +1,422 @@
+import com.hazelcast.jet.ProcessorSupplier;
+import com.hazelcast.jet.function.DistributedFunction;
+import com.hazelcast.jet.function.DistributedPredicate;
+import com.hazelcast.jet.impl.util.ExceptionUtil;
+import com.hazelcast.jet.processor.Processors;
+import com.hazelcast.journal.EventJournalInitialSubscriberState;
+import com.hazelcast.journal.EventJournalReader;
+import com.hazelcast.map.impl.journal.EventJournalMapEvent;
+import com.hazelcast.nio.Address;
+import com.hazelcast.projection.Projection;
+import com.hazelcast.ringbuffer.ReadResultSet;
+import com.hazelcast.ringbuffer.StaleSequenceException;
+import com.hazelcast.util.concurrent.BackoffIdleStrategy;
+import com.hazelcast.util.concurrent.IdleStrategy;
+
+import javax.annotation.Nonnull;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.IntStream;
+
+import static com.hazelcast.client.HazelcastClient.newHazelcastClient;
+import static com.hazelcast.jet.Util.entry;
+import static java.util.Collections.emptyList;
+import static java.util.concurrent.TimeUnit.MICROSECONDS;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.mapping;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.IntStream.range;
+
+public final class StreamEventJournalP<E, T> extends AbstractProcessor {
+
+    private static final IdleStrategy IDLER =
+            new BackoffIdleStrategy(0, 0, MICROSECONDS.toNanos(1), MILLISECONDS.toNanos(100));
+
+    private static final int MIN_FETCH_SIZE = 0;
+
+    private static final int MAX_FETCH_SIZE = 100;
+
+    private final EventJournalReader<E> eventJournalReader;
+
+    private final Map<Integer, Long> offsetMap;
+
+    private final SerializablePredicate<E> predicate;
+
+    private final Projection<E, T> projection;
+
+    private final boolean newestSequence;
+
+    private CompletableFuture<Void> jobFuture;
+
+    private StreamEventJournalP(EventJournalReader<E> eventJournalReader, List<Integer> partitions,
+                                Predicate<E> predicate,
+                                Function<E, T> projectionF,
+                                boolean newestSequence) {
+        this.eventJournalReader = eventJournalReader;
+        this.offsetMap = new HashMap<>();
+        this.predicate = predicate != null ? predicate::test : null;
+        this.projection = createProjection(projectionF);
+        this.newestSequence = newestSequence;
+        subscribe(partitions);
+
+    }
+
+    public static <T> ProcessorMetaSupplier streamMap(String mapName,
```

Comment:
can we move the static utility methods to the end of the file, as they interfere with the flow of the file